### PR TITLE
Update branch list for readthedocs configuration

### PR DESCRIPTION
**Pull request by priteau**
This will need to be merged into all newer branches.

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ build:
     python: "3.7"
   jobs:
     post_checkout:
-      - git remote set-branches origin master stackhpc/yoga stackhpc/xena stackhpc/wallaby
+      - git remote set-branches origin master stackhpc/2023.1 stackhpc/zed stackhpc/yoga stackhpc/xena stackhpc/wallaby
```
Comment (markgoddard) [inline .readthedocs.yaml:16]: Do the older branches need to know about later ones?

Comment (priteau) [reply .readthedocs.yaml:16]: I don't think so. Do you prefer the fix to be applied to each branch separately?

Comment (markgoddard) [reply .readthedocs.yaml:16]: I don't mind too much, but AFAICT each branch lists the earlier branches, so I was wondering if there is actually an issue to be fixed.

Comment (priteau) [reply .readthedocs.yaml:16]: I thought they didn't, I must have misread.

**Pull request closed without merging**